tests/sklearn/test_sklearn_models.py

```diff
@@ -2032,3 +2033,63 @@ def test_error_raise_unsupported_pandas_values(model_class, bad_value, expected_
 
     with pytest.raises(ValueError, match=expected_error):
         model.fit(x_train, y_train)
+
+
+@pytest.mark.parametrize(
+    "model_class, parameters",
+    get_sklearn_linear_models_and_datasets()
+    + get_sklearn_tree_models_and_datasets()
+    + get_sklearn_neighbors_models_and_datasets(),
+)
+def test_initialization_variables_match(model_class, parameters, load_data, is_weekly_option):
```

Comment:
should we also check the default values btw ? I know that we don't always have the same default values so this could be tricky to do, but then maybe add a fixme for that ?

Reply:
Good point! I think we should.